backend/src/test/kotlin/app/ehrenamtskarte/backend/cards/VerifyCardTest.kt

```diff
@@ -0,0 +1,173 @@
+            return listOf(
+                VerifyCardTestCase(
+                    createCard = ::staticValidCardWithoutExpirationDay,
+                    valid = true,
+                    extendable = false
+                ),
+                VerifyCardTestCase(
+                    createCard = ::staticExpiredExtendableCard,
+                    valid = false,
+                    extendable = true
+                ),
+                VerifyCardTestCase(
+                    createCard = ::staticFutureNonExtendableCard,
+                    valid = false,
+                    extendable = false
+                ),
+                VerifyCardTestCase(
+                    createCard = ::staticRevokedExtendableCard,
+                    valid = false,
+                    extendable = true
+                ),
+                VerifyCardTestCase(
+                    createCard = ::dynamicExtendableCardWithoutTotpSecret,
+                    valid = false,
+                    extendable = true
+                )
+            )
+        }
+
+        private fun staticValidCardWithoutExpirationDay(): Int {
+            val userId = TestData.createUserEntitlement(regionId = 95)
+            return TestData.createStaticCard(
+                entitlementId = userId,
+                expirationDay = null
+            )
+        }
+
+        private fun staticExpiredExtendableCard(): Int {
+            val userId = TestData.createUserEntitlement(regionId = 95, endDate = LocalDate.now().plusMonths(2L))
+            return TestData.createStaticCard(
+                entitlementId = userId,
+                expirationDay = LocalDate.now().minusMonths(1L).toEpochDay()
+            )
+        }
+
+        private fun staticFutureNonExtendableCard(): Int {
+            val userId = TestData.createUserEntitlement(regionId = 95, endDate = LocalDate.now().plusMonths(1L))
+            return TestData.createStaticCard(
+                entitlementId = userId,
+                startDay = LocalDate.now().plusMonths(1L).toEpochDay(),
+                expirationDay = LocalDate.now().plusMonths(2L).toEpochDay()
+            )
+        }
+
+        private fun staticRevokedExtendableCard(): Int {
+            val userId = TestData.createUserEntitlement(regionId = 95, endDate = LocalDate.now().plusMonths(2L))
+            return TestData.createStaticCard(
+                entitlementId = userId,
+                expirationDay = LocalDate.now().plusMonths(1L).toEpochDay(),
+                revoked = true
+            )
+        }
+
+        private fun dynamicExtendableCardWithoutTotpSecret(): Int {
+            val userId = TestData.createUserEntitlement(regionId = 95, endDate = LocalDate.now().plusMonths(2L))
+            return TestData.createDynamicCard(
+                entitlementId = userId,
+                expirationDay = LocalDate.now().plusMonths(1L).toEpochDay(),
+                totpSecret = null
+            )
+        }
+    }
+
+    @AfterEach
```

Comment:
:question: Wouldn't it make more sense to do this in `@BeforeEach` to ensure also the first test starts with a clean setup?

Reply:
actually yes, it would be more robust. will update that